cpu-sim/src/main.rs

```diff
@@ -19,6 +19,10 @@ struct Args {
     /// Enable verbose logging
     #[arg(short, long)]
     verbose: bool,
+
+    /// Enable debug logging (prints every instruction executed)
+    #[arg(short, long)]
+    debug: bool,
```

Comment:
Change this to something more descriptive. Let's use print-inst-trace instead since we're writing a log of all instructions run by the cpu.